Backend/app/kafka-consumers/src/Consumer/AvailabilityTime/Processor.hs

```diff
@@ -48,34 +50,66 @@ createOrUpdateDriverAvailability merchantId driverId (bucketStartTime, bucketEnd
-calculateAvailableTime merchantId driverId kc (fstTime : restTimeSeries, Just lastCR) = do
-  mbLatestAvailabilityRecord <- Q.findLatestByDriverIdAndMerchantId driverId merchantId
-  timeBetweenUpdates <- asks (.timeBetweenUpdates)
-  granualityPeriodType <- asks (.granualityPeriodType)
-  let lastAvailableTime = fromMaybe fstTime $ mbLatestAvailabilityRecord <&> (.lastAvailableTime)
-      activeTimePairs = mkPairsWithLessThenThreshold timeBetweenUpdates lastAvailableTime granualityPeriodType $ filter (> lastAvailableTime) (fstTime : restTimeSeries)
-      availabilityInWindow =
-        foldl
-          ( \acc timePair -> do
-              let bucketPair = getBucketPair granualityPeriodType timePair
-              flip (M.insert bucketPair) acc $
-                case M.lookup bucketPair acc of
-                  Just (oldAvailabileTime, _) -> (oldAvailabileTime + getTimeDiffInteger timePair, snd timePair)
-                  Nothing -> (getTimeDiffInteger timePair, snd timePair)
+calculateAvailableTime :: C.KafkaConsumer -> Map (T.MerchantId, T.DriverId) ([UTCTime], Maybe ConsumerRecordD) -> Flow ()
+calculateAvailableTime kafkaConsumer driverLocationUpdatesMap = do
+  void $ M.traverseWithKey calAvalTime driverLocationUpdatesMap
+  mapM_ (C.commitOffsetMessage C.OffsetCommit kafkaConsumer) getLatestCRInEachPartition
+  where
+    calAvalTime (driverId, merchantId) (timeSeries, _) =
+      calculateAvailableTime' merchantId driverId $ reverse timeSeries
+
+    getLatestCRInEachPartition =
+      map snd
+        . HM.elems
+        . foldl'
+          ( \patitionOffsetMap cr -> do
+              let paritionId = C.unPartitionId $ C.crPartition cr
+                  currentOffset = C.unOffset $ C.crOffset cr
+                  latestOffsetInPartition = HM.lookup paritionId patitionOffsetMap
+              if (latestOffsetInPartition <&> fst) > Just currentOffset
+                then patitionOffsetMap
+                else HM.insert paritionId (currentOffset, cr) patitionOffsetMap
           )
-          M.empty
-          activeTimePairs
-  logInfo $ "ActiveTime pairs " <> show activeTimePairs
-  logInfo $ "availabilityInWindow " <> show availabilityInWindow
-  void $ M.traverseWithKey (createOrUpdateDriverAvailability merchantId driverId) availabilityInWindow
-  void $ C.commitOffsetMessage C.OffsetCommit kc lastCR
+          HM.empty
+        . mapMaybe (snd . snd)
+        $ M.toList driverLocationUpdatesMap
+
+calculateAvailableTime' :: T.MerchantId -> T.DriverId -> [UTCTime] -> Flow ()
+calculateAvailableTime' _ _ [] = pure ()
+calculateAvailableTime' merchantId driverId (fstTime : restTimeSeries) = withLogTag driverId $
+  Redis.whenWithLockRedis (mkLastTimeStampKey <> "lock") 10 $ do
+    (unsureLastCommitedTime, isCommitedToDB) <- fromMaybe (fstTime, True) <$> Redis.get mkLastTimeStampKey
```

Comment:
Let's default `isCommitedToDB` to `False`.

Also spell check -> `isCommittedToDB`